json-rpc/tests/jsonrpc-integration-tests/main.rs

```diff
@@ -0,0 +1,2098 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_crypto::{hash::CryptoHash, HashValue};
+use diem_json_rpc::views::{
+    AccountTransactionsWithProofView, AccumulatorConsistencyProofView, EventView,
+};
+use diem_sdk::{transaction_builder::Currency, types::AccountKey};
+use diem_transaction_builder::stdlib;
+use diem_types::{
+    access_path::AccessPath,
+    account_address::AccountAddress,
+    account_config::{treasury_compliance_account_address, xus_tag},
+    contract_event::EventWithProof,
+    diem_id_identifier::DiemIdVaspDomainIdentifier,
+    event::EventKey,
+    ledger_info::LedgerInfoWithSignatures,
+    on_chain_config::DIEM_MAX_KNOWN_VERSION,
+    proof::{AccumulatorConsistencyProof, TransactionAccumulatorSummary},
+    transaction::{
+        AccountTransactionsWithProof, ChangeSet, Transaction, TransactionPayload, WriteSetPayload,
+    },
+    write_set::{WriteOp, WriteSet, WriteSetMut},
+};
+use forge::{
+    forge_main, AdminContext, AdminTest, ForgeConfig, LocalFactory, Options, PublicUsageContext,
+    PublicUsageTest, Result, Test,
+};
+use serde_json::json;
+use std::{
+    convert::{TryFrom, TryInto},
+    ops::Deref,
+    str::FromStr,
+};
+
+mod helper;
+use helper::JsonRpcTestHelper;
+
+fn main() -> Result<()> {
+    let tests = ForgeConfig {
+        public_usage_tests: &[
+            &CurrencyInfo,
+            &BlockMetadata,
+            &OldMetadata,
+            &AccoutNotFound,
+            &UnknownAccountRoleType,
+            &DesignatedDealerPreburns,
+            &ParentVaspAccountRole,
+            &GetAccountByVersion,
+            &ChildVaspAccountRole,
+            &PeerToPeerWithEvents,
+            &PeerToPeerErrorExplination,
+            &ReSubmittingTransactionWontFail,
+            &MempoolValidationError,
+            &ExpiredTransaction,
+            &RotateComplianceKeyEvent,
+            &CreateAccountEvent,
+            &GetTransactionsWithoutEvents,
+            &GetAccountTransactionsWithoutEvents,
+            &GetAccountTransactionsWithProofs,
+            &GetTransactionsWithProofs,
+            &GetTreasuryComplianceAccount,
+            &GetEventsWithProofs,
+            &MultiAgentPaymentOverDualAttestationLimit,
+            &GetAccumulatorConsistencyProof,
+            &NoUnknownEvents,
+        ],
+        admin_tests: &[
+            &PreburnAndBurnEvents,
+            &CancleBurnEvent,
+            &UpdateExchangeRateEvent,
+            &MintAndReceivedMintEvents,
+            &AddAndRemoveDiemIdDomain,
+            &MultiAgentRotateAuthenticationKeyAdminScript,
+            &MultiAgentRotateAuthenticationKeyAdminScriptFunction,
+            &UpgradeEventAndNewEpoch,
+            &UpgradeDiemVersion,
+        ],
+        network_tests: &[],
+    };
+
+    let options = Options::from_args();
+    forge_main(tests, LocalFactory::from_workspace()?, &options)
```

Comment:
share we migration all this part under Forge itself instead of create new configs and forge instance separately to run tests

Reply:
I think given these tests are specific to jsonrpc I think it makes more sense for them to live there. The nice thing about these tests are they're just types that implement the interfaces so what we might end up with is sticking these tests in a crate and then publicly exporting the tests such that they can run in a local harness like this or included in a testsuite that is run by the forge binary